File: frontend/src/features/news/types/article.ts
export interface Article {
  id: number;

  title: string;

  summary: string | null;

  content: string | null;

  url: string;

  image_url: string | null;

  source: string;

  author: string | null;

  language: string;

  country: string;

  category: string;

  published_at: string | null;

  created_at: string;

  updated_at: string;
}

File: backend/app/schemas/article.py
from datetime import datetime

from pydantic import BaseModel


class ArticleResponse(BaseModel):
    id: int
    title: str
    summary: str | None
    url: str
    source: str
    author: str | None
    language: str
    country: str
    category: str
    published_at: datetime | None

    model_config = {
        "from_attributes": True,
    }